procurement_purchase_no_grouping/models/stock_rule.py

```diff
@@ -23,13 +24,18 @@ def _run_buy(self, procurements):
     def _make_po_get_domain(self, company_id, values, partner):
         """Inject an impossible domain for not getting match in case of no order
         grouping.
-
         We try to make it the more unique possible for avoiding coincidences for not
         overlapping in a batch procurement run (like a sales order with multiple MTO
         lines confirmation).
         """
         domain = super()._make_po_get_domain(company_id, values, partner)
-        if values.get("grouping") == "order":
+        if values.get("grouping") == "product_category":
+            if values.get("product_id"):
+                product = self.env["product.product"].browse(values["product_id"])
+                domain += (
+                    ("order_line.product_id.categ_id", "=", product.categ_id.id),
+                )
+        elif values.get("grouping") == "order":
```

Comment:
In the worst case, only take the value once
```suggestion
        grouping = values.get("grouping")
        if grouping == "product_category":
            if values.get("product_id"):
                product = self.env["product.product"].browse(values["product_id"])
                domain += (
                    ("order_line.product_id.categ_id", "=", product.categ_id.id),
                )
        elif grouping == "order":
```

Reply:
This barely optimizes, as both is an access to a variable or a value in a dictionary. There are microseconds of difference. It's no the same as caching other things in variables that requires a more complicated access (ORM fields for example), or for clearer code.